object_intern.go

```diff
@@ -100,49 +103,88 @@ func (oi *ObjectIntern) DecompressString(in string) (string, error) {
 // If the object is found in the store its reference count is increased by 1.
 // If the object is added to the store its reference count is set to 1.
 func (oi *ObjectIntern) AddOrGet(obj []byte, safe bool) (uintptr, error) {
-	var addr gos.ObjAddr
-	var ok bool
-	var err error
-	var objString string
 
-	objComp := obj
+	// if either of these two terms is true then the rest of this block
+	// requires a lot of allocations
+	if (oi.conf.Compression != None) || (safe && oi.conf.Compression == None) {
+		objComp := obj
 
-	if oi.conf.Compression != None {
-		objComp = oi.compress(obj)
-	}
+		if oi.conf.Compression != None {
+			// this returns a new byte slice, so we don't need to check for safe
+			objComp = oi.compress(obj)
+		}
 
-	// the only case we need to handle specially is when compression
-	// is turned off and the user has requested that the operation be safe
-	if safe && oi.conf.Compression == None {
-		// create a copy so we don't modify the original []byte
-		objComp = make([]byte, len(obj))
-		copy(objComp, obj)
-	}
+		// the only case we need to handle specially is when compression
+		// is turned off and the user has requested that the operation be safe
+		if safe && oi.conf.Compression == None {
```

Comment:
this could be an `else` of the previous block

Reply:
Yes, I will look at refactoring that a little bit to avoid the allocation, based on your above comment.